src/TemplatesBuilder.js

```diff
@@ -224,7 +225,10 @@ export default class TemplatesBuilder {
       propertyTemplate.valueSubjectTemplateKeys = this.valuesFor(
         attributeTerm,
         "http://sinopia.io/vocabulary/hasResourceTemplateId"
-      )
+      ).map((subjectTemplateKey) => resourceToName(subjectTemplateKey))
```

Comment:
Keys get more complicated for nested resource properties in order to make them unique.